test/test_io.py

```diff
@@ -2681,6 +2682,44 @@ def test_mean_axis(self):
         self.assertIsInstance(m, FlowCal.io.FCSData)
         self.assertEqual(m.shape, m_array.shape)
         np.testing.assert_array_equal(m, m_array)
-        
+
+class TestFCSDataPickle(unittest.TestCase):
+    '''See Issue #277. FCSData objects must be able to be serialized (via Pickle)
+    and deserialized correctly.'''
+
+    def setUp(self):
+        self.d = FlowCal.io.FCSData(filenames[2])
+        file_dir = os.path.dirname(os.path.abspath(__file__))
+        self.test_file = os.path.join(file_dir, 'test_fcs.p')
+
+    def test_pickle_unpickle(self):
+        """
+        Test that an FCSData instance can be pickled, then unpickled, and
+        still retain all custom attributes.
+        """
+        channels = self.d.channels # Should be fine, no error
+
+        with open(self.test_file, 'w') as outfile:
+            cPickle.dump(self.d, outfile) # serialization
+
+        with open(self.test_file, 'r') as infile:
+            test_fcs = cPickle.load(infile) # de-serialization
+
+        # Check all attributes/methods are preserved
+        attrs = set(dir(self.d))
+        loaded_attrs = set(dir(test_fcs))
+        self.assertEqual(attrs, loaded_attrs)
+        # Check contents of attrs affected by pickling
+        pickle_sensitive_attrs = ['_resolution', '_range', '_amplifier_gain',
+            '_detector_voltage', '_amplification_type', '_channels',
+            '_acquisition_end_time', '_acquisition_start_time', '_time_step'
+            '_data_type', '_analysis', '_text', '_infile']
+        for attr in pickle_sensitive_attrs:
+            if hasattr(self.d, attr):
```

Comment:
Why is this necessary? You should know in advance if these attributes are valid or not.

Reply:
Yep, you're right - this is now deleted.